### PR TITLE
Tag users on goals

### PR DESCRIPTION
**Pull request by RubenSousaDinis**
## Summary

This PR adds a way to tag users on goals

### CODE REVIEW

File: app/services/goals/update.rb [hunk 1]
```diff
@@ -34,6 +37,36 @@ def call
 
     attr_reader :goal, :params
 
+    def upsert_accountability_users
+      raise UpdateError, "You can only add 10 accountability buddies" if accountability_users.count > 10
+
+      goal.goal_accountability_users.update_all(removed: true)
+      accountability_users.each do |user|
+        goal_accountability_user = GoalAccountabilityUser.find_by(user: user, goal: goal)
+        goal_accountability_user ||= GoalAccountabilityUser.new(user: user, goal: goal)
```
Comment (francisco-leal) [inline app/services/goals/update.rb:46]: you can use `find_or_initialize_by`

File: app/blueprints/api/goal_blueprint.rb [hunk 2]
```diff
@@ -28,5 +29,9 @@ class API::GoalBlueprint < Blueprinter::Base
         goal.election.votes.where(candidate: goal.user, wallet_id: options[:current_user].wallet_id).sum(&:amount)
       end
     end
+
+    field :accountability_users_count do |goal, options|
+      goal.accountability_users.count
```
Comment (francisco-leal) [inline app/blueprints/api/goal_blueprint.rb:34]: aren't we missing a scope here due to the `removed` field?

File: app/services/goals/update.rb [hunk 3]
```diff
@@ -34,6 +37,36 @@ def call
...
+        goal_accountability_user.update!(removed: false)
+      end
+    end
+
+    def accountability_users
+      User.where(uuid: params[:accountability_users])
+    end
+
+    def notify_accountability_users
+      goal.goal_accountability_users.where(notified_at: nil).find_each do |accountability_user|
```
Comment (francisco-leal) [inline app/services/goals/update.rb:57]: aren't we missing a scope to filter out removed users?

File: app/services/goals/update.rb (hunk 1 above)
Comment (RubenSousaDinis) [reply app/services/goals/update.rb]: changed

File: app/blueprints/api/goal_blueprint.rb (hunk 2 above)
Comment (RubenSousaDinis) [reply app/blueprints/api/goal_blueprint.rb]: good spot. Changed

File: app/services/goals/update.rb (hunk 3 above)
Comment (RubenSousaDinis) [reply app/services/goals/update.rb]: Yes, changed

**Pull request merged**